ecs-cli/modules/cli/servicediscovery/servicediscovery_app.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2015-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package servicediscovery
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/cloudformation"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands/flags"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+const (
+	privateDNSNamespaceStackNameFormat     = "amazon-ecs-cli-setup-%s-%s-private-dns-namespace"
+	serviceDiscoveryServiceStackNameFormat = "amazon-ecs-cli-setup-%s-%s-service-discovery-service"
+)
+
+// CloudFormation template parameters
+const (
+	parameterKeyNamespaceDescription = "NamespaceDescription"
+	parameterKeyVPCID                = "VPCID"
+	parameterKeyNamespaceName        = "NamespaceName"
+)
+
+const (
+	parameterKeySDSDescription                          = "SDSDescription"
+	parameterKeySDSName                                 = "SDSName"
+	parameterKeyNamespaceID                             = "NamespaceID"
+	parameterKeyDNSType                                 = "DNSType"
+	parameterKeyDNSTTL                                  = "DNSTTL"
+	parameterKeyHealthCheckCustomConfigFailureThreshold = "FailureThreshold"
+)
+
+const (
+	CFNTemplateOutputPrivateNamespaceID = "PrivateDNSNamespaceID"
+	CFNTemplateOutputSDSARN             = "ServiceDiscoveryServiceARN"
+)
+
+const (
+	dnsRecordTypeA   = "A"
+	dnsRecordTypeSRV = "SRV"
+)
+
+var requiredParamsSDS = []string{parameterKeyNamespaceID, parameterKeySDSName, parameterKeyDNSType}
+var requiredParamsNamespace = []string{parameterKeyVPCID, parameterKeyNamespaceName}
+
+// Create creates resources for service discovery and returns the ID of the Service Discovery Service
+func Create(c *cli.Context, networkMode, serviceName, clusterName string) (*string, error) {
```

Comment:
Could you refresh my memory -- I was under the impression from the design discussion that there would not be a separate/public Create command to only create Service Discovery resources?

Reply:
Yeah, there isn't a separate command. This function is called from compose, but for the sake of keeping this stuff out of the mess of compose packages that we have, I gave it its own package at this level. Do you think this is okay?